skbio/sequence/_nucleotide_mixin.py

```diff
@@ -161,6 +161,15 @@ def complement(self, reverse=False):
 
         if reverse:
             complement = complement[::-1]
+
+        if self.has_interval_metadata():
+            interval_metadata = self.interval_metadata
```

Comment:
`IntervalMetadata._reverse` mutates the object in-place. This method produces a new `Sequence` with copies of the reversed metadata, so we'd want to first do a shallow-copy of `self.interval_metadata` and reverse the copy.

To achieve this, I think the code can be simplified while also matching the above pattern used for `metadata` and `positional_metadata`. It will also be more efficient by avoiding an extra shallow-copy operation:

```python
interval_metadata = None
if self.has_interval_metadata():
    interval_metadata = self.interval_metadata

complement = self._constructor(
    sequence=result,
    metadata=metadata,
    positional_metadata=positional_metadata,
    interval_metadata=interval_metadata)

if reverse:
    complement = complement[::-1]

    # TODO: this can be revised to match
    # positional_metadata when __getitem__
    # supports interval_metadata
    if complement.has_interval_metadata():
        complement.interval_metadata._reverse()
```

Can you also add a unit test that would have caught the bug?